hi-ml-multimodal/src/health_multimodal/image/model/model.py

```diff
@@ -87,6 +87,7 @@ def __init__(self,
 
         # Initiate encoder, projector, and classifier
         self.encoder = ImageEncoder(img_model_type)
+        self.encoder.eval()  # to get the output size in the next line
```

Comment:
By convention `nn.modules` are initialised with `self.training` set to `True`.

Instead of hardcoding this here, perhaps we could handle this inside the `get_encoder_output_dim` method also in case other classes use this function later.

One idea is that -- In line 214, you could add the following:
```
        initial_state = self.encoder.training
        module.eval()
        module.forward(...)
        module.train(initial_state) # Restore the state of encoder again
```

Reply:
Thanks, I thought of doing that, potentially with a context manager. But I think in this case it shouldn't be necessary as we have `self.train()` in the `__init__` of the class, just a few lines below: https://github.com/microsoft/hi-ml/pull/809/files#diff-ddbcf0079364b4612521f029a11af075c8293ec9c06f147e9cedafa17f6c42bdR99

Reply:
I've created a context manager in [4bdc2e7](https://github.com/microsoft/hi-ml/pull/809/commits/4bdc2e7bf8cd61bb53431c827955679229e21b4c) to handle this in a generic way.

Reply:
Looks good to me, if we remove this line and use the context manager you created instead.

Reply:
Thanks! I forgot to remove it. Done in 9533bead.